lib/ansible_tower_client/api.rb

```diff
@@ -50,7 +52,18 @@ def job_templates
     end
 
     def method_missing(method_name, *args, &block)
-      instance.respond_to?(method_name) ? instance.send(method_name, *args, &block) : super
+      if instance.respond_to?(method_name)
+        logger.debug("#{self.class.name} Sending <#{method_name}> with <#{args.inspect}>") if logger.debug?
+        instance.send(method_name, *args, &block).tap do |response|
+          if logger.debug?
+            require 'pp'
```

Comment:
Thinking this `require` should be moved out of the method so it is not hit on every call but only once when the class is loaded.  I realized it will only be run when debug logging is enabled, but still seem better to process it one time.